libcontainer/nsenter/nsenter_test.go

```diff
@@ -49,6 +46,7 @@ func TestNsenterValidPaths(t *testing.T) {
 	if err := cmd.Start(); err != nil {
 		t.Fatal("nsenter failed to start:", err)
 	}
+	child.Close()
```

Comment:
should we use a `defer` so that we can set that directly after creating it (with newpipe)?

I guess this is the "Make sure we close child fds after the start." case? Is the Cleanup() not enough for this? (curious)

Reply:
Not sure I understand the question.

We create a pipe, one end is for the parent, the other is for the child. Once we have forked, the child end needs to be closed. This explicit close is needed because otherwise we won't get EOF in the parent once the child is gone (as the pipe will still be open).

Cleanup() is only executed after the test has finished.

Reply:
The part about EOF is irrelevant as we don't need an EOF from the child. In this case we close the child end of pipe right after the fork merely because we no longer need it, and it's always good to do it explicitly as at the very least it improves code readability ("we passed this end of the pipe to the child and closed it right after, as we have no use for it").

Reply:
> In this case we close the child end of pipe right after the fork merely because we no longer need it,

gotcha, yes, makes sense (was merely looking; do we need to do so, because `t.Cleanup()` also does so